### PR TITLE
[HWArith][HWArithToHW] Implement ICmp

### PR DESCRIPTION
**Pull request by 7FM**
This PR adds the missing comparison operator (https://github.com/llvm/circt/issues/3549) as well as implements the lowering to `comb`. I can also split this PR into two separate ones, if preferred.

### CODE REVIEW

File: test/Conversion/HWArithToHW/test_basic.mlir [hunk 1]
```diff
@@ -213,3 +213,41 @@ hw.module @div(%op0: i32, %op1: i32) -> (sisi: i32, siui: i32, uisi: i32, uiui:
 // CHECK:   hw.output %[[SISI_OUT]], %[[SIUI_RES]], %[[UISI_OUT]], %[[UIUI_RES]] : i32, i32, i32, i32
   hw.output %sisiOut, %siuiOut, %uisiOut, %uiuiOut : i32, i32, i32, i32
 }
+
+// -----
+
+// CHECK: hw.module @icmp(%op0: i32, %op1: i32) -> (sisi: i1, siui: i1, uisi: i1, uiui: i1) {
+hw.module @icmp(%op0: i32, %op1: i32) -> (sisi: i1, siui: i1, uisi: i1, uiui: i1) {
```
Comment (mortbopet) [inline test/Conversion/HWArithToHW/test_basic.mlir:220]: Adding some comparisons on mixed-width, mixed sign values would be good

File: include/circt/Dialect/HWArith/HWArithOps.h [hunk 2]
```diff
@@ -22,4 +22,13 @@
 #define GET_OP_CLASSES
 #include "circt/Dialect/HWArith/HWArith.h.inc"
 
+namespace circt {
+namespace hwarith {
+
+unsigned inferAddResultType(IntegerType::SignednessSemantics &signedness,
```
Comment (mortbopet) [inline include/circt/Dialect/HWArith/HWArithOps.h:28]: Nit: Missing comment

File: lib/Conversion/HWArithToHW/HWArithToHW.cpp [hunk 3]
```diff
@@ -167,6 +167,61 @@ struct CastOpLowering : public OpConversionPattern<CastOp> {
 };
 } // namespace
 
+namespace {
+static comb::ICmpPredicate lowerPredicate(ICmpPredicate pred, bool isSigned) {
```
Comment (mortbopet) [inline lib/Conversion/HWArithToHW/HWArithToHW.cpp:171]: Nit: Missing comment

**Review by mortbopet — APPROVED** [review]
A few nits + some extra tests, but otherwise LGTM, thanks!

**Comment by 7FM** [comment]
> A few nits + some extra tests, but otherwise LGTM, thanks!

Thanks for your reviews :)

**Pull request merged**